src/main/java/com/kainos/ea/backend/services/JobFamilyService.java

```diff
@@ -1,24 +1,46 @@
 package com.kainos.ea.backend.services;
 
 import com.kainos.ea.backend.models.JobFamily;
+import com.kainos.ea.backend.repositories.CapabilityRepository;
 import com.kainos.ea.backend.repositories.JobFamilyRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import javax.management.InstanceAlreadyExistsException;
+import javax.naming.InvalidNameException;
 import java.util.List;
 
 @Service
 public class JobFamilyService {
 
     private JobFamilyRepository jobFamilyRepository;
+    private CapabilityService capabilityService;
 
     @Autowired
-    public JobFamilyService(JobFamilyRepository jobFamilyRepository) {
+    public JobFamilyService(JobFamilyRepository jobFamilyRepository, CapabilityService capabilityService) {
         this.jobFamilyRepository = jobFamilyRepository;
+        this.capabilityService = capabilityService;
     }
 
     public List<JobFamily> getJobFamiliesByCapabilityName(String capabilityName) {
         return jobFamilyRepository.findByCapabilityName(capabilityName);
     }
 
+    public JobFamily addJobFamily(JobFamily jobFamily) throws InvalidNameException, InstanceAlreadyExistsException {
+        if (!capabilityService.capabilityExists(jobFamily.getCapability().getName()))
```

Comment:
These if statements maybe could be separated into a function like validateJobFamily?